subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/ProjectExtensions.kt

```diff
@@ -227,7 +229,18 @@ fun Project.gradleKotlinDsl(): Dependency =
     DefaultSelfResolvingDependency(
         fileCollectionOf(
             gradleKotlinDslOf(project),
-            "gradleKotlinDsl") as FileCollectionInternal)
+            "gradleKotlinDsl"
+        ) as FileCollectionInternal
+    )
+
+
+fun Project.gradleKotlinDslJars(): FileCollection =
+    fileCollectionOf(
+        project.serviceOf<KotlinScriptClassPathProvider>().gradleKotlinDslJars.asFiles.filter {
```

Comment:
:o: I would extract that part to `KotlinScriptClassPathProvider.kt` so it can reuse the already existing private helpers there